specs/IFrameMemoryUsage.md

```diff
@@ -0,0 +1,306 @@
+<pre>
+                           main_frame  [depth:0]
+                         /     |       \
+      first_level_iframe_0     ...    first_level_iframe_n             [depth:1, first level iframe]
+                |          \                     |
+      second_level_iframe_0 ...       second_level_iframe_n ...        [depth:2, second level iframe]
+               |
+              ...                                                      
+</pre> 
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` and `CoreWebView2Frame` to include the 
+`RendererProcessId` property and  `RendererProcessIdChanged` event. This property 
+stores the renderer process ID for its iframe. This event will be raised whenever 
+the renderer process ID is changed for `CoreWebView2` or `CoreWebView2Frame`.
+
+Additionally, we propose to add the `FrameCreated` event to `CoreWebView2Frame`. 
+This event is raised when a new iframe is created.
+
+# Examples
+C++
+```c++
+void InitializeEventView(wil::com_ptr<ICoreWebView2> webview)
+{
+    auto webview2_18 = webview.try_query<ICoreWebView2_18>();
+    if (webview2_18) 
+    {
+        webview2_18->add_FrameCreated(
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [this](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args)
+                    -> HRESULT {
+                    // ...
+                    // The new frame created event can be recursively initialized
+                    InitializeFrameEventView(webviewFrame);
+                    // ...
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+        //! [ICoreWebView2:add_RendererProcessIdChanged]
+        webview2_18->add_RendererProcessIdChanged(
+            Callback<ICoreWebView2RendererProcessIdChangedEventHandler>(
+                [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT
+                {
+                    wil::com_ptr<ICoreWebView2_18> webview2_18;
+                    sender->QueryInterface(IID_PPV_ARGS(&webview2_18));
+                    UINT32 processId;
+                    //! [ICoreWebView2:get_RendererProcessId]
+                    CHECK_FAILURE(webview2_18->get_RendererProcessId(&processId));
+                    // Do something when renderer process id has been changed. 
+                    // For instance, retrieve the updated memory usage information 
+                    // for that main frame. 
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+    }
+}
+
+void InitializeFrameEventView(
+    wil::com_ptr<ICoreWebView2Frame> webviewFrame) 
+{
+    auto frame5 = webviewFrame.try_query<ICoreWebView2Frame5>();
+    if (frame5)
+    {
+        //! [ICoreWebView2Frame:add_FrameCreated]
+        frame5->add_FrameCreated(
+            Callback<ICoreWebView2NestedFrameCreatedEventHandler>(

```

Comment:
`NestedFrameCreated` isn't following naming convention, but maybe it's OK; naming convention would be `FrameFrameCreated`.

Reply:
Yes, please use: ICoreWebView2ChildFrameCreatedEventHandler